internal/controllers/machineset/machineset_controller.go

```diff
@@ -1604,3 +1602,21 @@ func (r *Reconciler) reconcileExternalTemplateReference(ctx context.Context, clu
 
 	return false, patchHelper.Patch(ctx, obj)
 }
+
+// Returns the machines to be remediated in the following order
+//   - Machines with RemediateMachineAnnotation annotation if any,
+//   - Machines failing to come up first because
+//     there is a chance that they are not hosting any workloads (minimize disruption).
+func sortMachinesToRemediate(machines []*clusterv1.Machine) {
+	sort.SliceStable(machines, func(i, j int) bool {
+		_, iHasRemediateAnnotation := machines[i].Annotations[clusterv1.RemediateMachineAnnotation]
+		_, jHasRemediateAnnotation := machines[j].Annotations[clusterv1.RemediateMachineAnnotation]
+		if iHasRemediateAnnotation && !jHasRemediateAnnotation {
+			return true
+		}
+		if !iHasRemediateAnnotation && jHasRemediateAnnotation {
+			return false
+		}
```

Comment:
Up to you if to apply the change, but for sake of consistency/readability we can rewrite this as
```suggestion
	if annotations.HasRemediateMachine(i) && !annotations.HasRemediateMachine(j) {
		return true
	}
	if !annotations.HasRemediateMachine(i) && annotations.HasRemediateMachine(j) {
		return false
	}
```

Reply:
Updated as it makes more simpler.